Automatic TLS support in new Vert.x based open telemetry implementation

### Describe the bug

We have a quarkus application with open telemetry enabled, using an https OLTP endpoint for exporting traces. 
Up until Quarkus 3.2.x, the OkHttp-based implementation was handling TLS connections automatically without requiring any special configuration.
Once we upgraded to the 3.3.x branch, we started getting the following error when otel was attempting to export the traces:

```
Failed to export spans. The request could not be executed. Full error message: Failed to create SSL connection: javax.net.ssl.SSLHandshakeException: Failed to create SSL connection
        at io.vertx.core.net.impl.ChannelProvider$1.userEventTriggered(ChannelProvider.java:127)
        at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:400)
        at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:376)
        at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:368)
        at io.netty.handler.ssl.SslHandler.handleUnwrapThrowable(SslHandler.java:1270)
        at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1251)
        at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1295)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:101)
        at java.base/sun.security.validator.Validator.getInstance(Validator.java:181)
        at java.base/sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:309)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:183)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:255)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:144)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1335)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1232)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1175)
        at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:396)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:480)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1277)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1264)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:1209)
        at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1559)
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1405)
        at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1246)
        ... 20 more
Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
        at java.base/java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
        at java.base/java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
        at java.base/java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
        at java.base/sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:98)
        ... 37 more
```

(also reported in https://github.com/quarkusio/quarkus/issues/35786, but it turns out it was a different issue)

I don't know if this is a bug report or a feature request, but Quarkus should be able to handle simple https OLTP endpoints without extra configuration.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Use the quick start project for OTEL: https://github.com/quarkusio/quarkus-quickstarts/tree/main/opentelemetry-quickstart
set a dummy https OTLP endpoint, e.g.
```
quarkus.otel.exporter.otlp.traces.endpoint=https://github.com
```
hit `http://localhost:8080/hello` and you will get the error log

### Output of `uname -a` or `ver`

Darwin K4V32WY17P 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 LTS OpenJDK Runtime Environment Corretto-17.0.5.8.1 (build 17.0.5+8-LTS) OpenJDK 64-Bit Server VM Corretto-17.0.5.8.1 (build 17.0.5+8-LTS, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.3.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java
+++ b/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OtlpRecorder.java
@@ -219,8 +219,12 @@ public class OtlpRecorder {
         }
 
         private KeyCertOptions toPemKeyCertOptions() {
-            PemKeyCertOptions pemKeyCertOptions = new PemKeyCertOptions();
             OtlpExporterTracesConfig.KeyCert keyCert = tracesConfig.keyCert();
+            if (keyCert.certs().isEmpty() && keyCert.keys().isEmpty()) {
+                return null;
+            }
+
+            PemKeyCertOptions pemKeyCertOptions = new PemKeyCertOptions();
             if (keyCert.certs().isPresent()) {
                 for (String cert : keyCert.certs().get()) {
                     pemKeyCertOptions.addCertPath(cert);
@@ -235,14 +239,18 @@ public class OtlpRecorder {
         }
 
         private PemTrustOptions toPemTrustOptions() {
-            PemTrustOptions pemTrustOptions = new PemTrustOptions();
             OtlpExporterTracesConfig.TrustCert trustCert = tracesConfig.trustCert();
             if (trustCert.certs().isPresent()) {
-                for (String cert : trustCert.certs().get()) {
-                    pemTrustOptions.addCertPath(cert);
+                List<String> certs = trustCert.certs().get();
+                if (!certs.isEmpty()) {
+                    PemTrustOptions pemTrustOptions = new PemTrustOptions();
+                    for (String cert : trustCert.certs().get()) {
+                        pemTrustOptions.addCertPath(cert);
+                    }
+                    return pemTrustOptions;
                 }
             }
-            return pemTrustOptions;
+            return null;
         }
     }
 }

